frontend/src/metabase/static-viz/components/PieChart/setttings.ts

```diff
@@ -0,0 +1,74 @@
+import type {
+  ComputedVisualizationSettings,
+  RemappingHydratedDatasetColumn,
+} from "metabase/visualizations/types";
+import type {
+  DatasetColumn,
+  PieChartSettings,
+  RawSeries,
+  VisualizationSettings,
+} from "metabase-types/api";
+import { getDefaultDimensionAndMetric } from "metabase/visualizations/lib/utils";
+import { getColorsForValues } from "metabase/lib/colors/charts";
+import { normalize } from "metabase-lib/queries/utils/normalize";
+import { getColumnKey } from "metabase-lib/queries/utils/get-column-key";
+
+import { SLICE_THRESHOLD } from "./constants";
+
+// TODO move this to base branch in shared location
+const getColumnSettings = (
+  column: DatasetColumn,
+  settings: VisualizationSettings,
+) => {
+  const columnKey = Object.keys(settings.column_settings ?? {}).find(
+    possiblyDenormalizedFieldRef =>
+      normalize(possiblyDenormalizedFieldRef) === getColumnKey(column),
+  );
+  if (!columnKey) {
+    return null;
+  }
+  return settings.column_settings?.[columnKey];
+};
+
+function getPieChartColors(
+  rawSeries: RawSeries,
+  currentSettings: Partial<PieChartSettings>,
+): PieChartSettings["pie.colors"] {
+  const [
+    {
+      data: { rows, cols },
+    },
+  ] = rawSeries;
+
+  const dimensionIndex = cols.findIndex(
+    col => col.name === currentSettings["pie.dimension"],
+  );
+  const dimensionValues = rows.map(r => String(r[dimensionIndex]));
+
+  return getColorsForValues(dimensionValues, currentSettings["pie.colors"]);
+}
+
+export function computeStaticPieChartSettings(
+  rawSeries: RawSeries,
+): ComputedVisualizationSettings {
+  const settings: ComputedVisualizationSettings = {
+    ...rawSeries[0].card.visualization_settings,
+  };
+
+  if (!settings["pie.dimension"] || !settings["pie.metric"]) {
+    const defaults = getDefaultDimensionAndMetric(rawSeries);
+    settings["pie.dimension"] ??= defaults.dimension;
+    settings["pie.metric"] ??= defaults.metric;
+  }
+
+  settings["pie.show_legend"] ??= true;
```

Comment:
Wdyt about extracting the default value computation into separate functions that are reused both in the settings definitions and here? I tried to follow this approach here https://github.com/metabase/metabase/pull/34763/files#diff-8b242d67ce06af681f4568ad6bfbe2b55d76dc11838c0022c68bec3c05f368cdR32 
It may look goofy for such simple cases but I believe it adds resilience and eliminates the possibility of updating the settings definition only while forgetting about static charts

Reply:
@EmmadUsmani what is your opinion on this?